Model/Api/Tax.php

```diff
@@ -169,17 +181,53 @@ public function createInstorePickOption($totalsInformation, $currencyCode, $ship
         $storeAddress->setRegion($ship_to_store_option['address']['region']);
         $storeAddress->setPostalCode($ship_to_store_option['address']['postal_code']);
         
+        $ship_to_store_option = $this->getShippingDiscount($totalsInformation, $currencyCode, $ship_to_store_option);
+        
         $shipToStoreOption = $this->shipToStoreOptionFactory->create();
         $shipToStoreOption->setReference($ship_to_store_option['reference']);
-        $shipToStoreOption->setCost(CurrencyUtils::toMinor($totalsInformation->getShippingAmount(), $currencyCode));
+        $shipToStoreOption->setCost($ship_to_store_option['cost']);
         $shipToStoreOption->setStoreName($ship_to_store_option['store_name']);
         $shipToStoreOption->setAddress($storeAddress);
         $shipToStoreOption->setDistance($ship_to_store_option['distance']);
         $shipToStoreOption->setDistanceUnit($ship_to_store_option['distance_unit']);
-        $shipToStoreOption->setTaxAmount(CurrencyUtils::toMinor($totalsInformation->getShippingTaxAmount(), $currencyCode));
+        $shipToStoreOption->setTaxAmount($ship_to_store_option['tax_amount']);
 
         return $shipToStoreOption;
     }
+    
+    /**
+     * @param TotalsInterface $totalsInformation
+     * @param string $currencyCode
+     * @param array $shipping_option
```

Comment:
Let's stick to camel case whenever possible

Reply:
Fixed